session/payment/session_balance.go

```diff
@@ -97,9 +124,88 @@ func (ppo *SessionBalance) Start() error {
 	}
 }
 
+func (ppo *SessionBalance) loadInitialPromiseState() (promise.StoredPromise, error) {
+	lastPromise, err := ppo.promiseStorage.GetLastPromise(ppo.issuer)
+	if err != nil {
+		// if an error occurs when fetching the last promise, issue a new id
+		ppo.sequenceID, err = ppo.promiseStorage.GetNewSeqIDForIssuer(ppo.issuer)
+		if err != nil {
+			return lastPromise, err
+		}
+	} else {
+		ppo.sequenceID = lastPromise.SequenceID
+	}
+	return lastPromise, nil
+}
+
+func (ppo *SessionBalance) startBalanceTracker(lastPromise promise.StoredPromise) {
+	amountToAdd := lastPromise.UnconsumedAmount
+
+	ppo.balanceTracker.Add(amountToAdd)
+	ppo.balanceTracker.Start()
+}
+
+func (ppo *SessionBalance) calculateUnconsumedPromiseAmount(balance uint64, lastPromise promise.StoredPromise) (uint64, error) {
+	// if we're ever in a situation where the unconsumed amount is zero, but the balance is not - something is definitely not right
+	if lastPromise.UnconsumedAmount == 0 && balance != 0 {
+		return 0, fmt.Errorf("unconsumed amount is 0, while balance is %v", balance)
+	}
+	unconsumed := lastPromise.UnconsumedAmount - (lastPromise.UnconsumedAmount - balance)
```

Comment:
It's equal to:
```golang
	unconsumed := balance
```
Probably something wrong here.

Reply:
I actually LOL'ed. Leftover from extensive rework of this part, will fix.